test/api_test.rb

```diff
@@ -67,7 +69,9 @@
       exception = assert_raises ArgumentError do
         Asciidoctor.load_file(sample_input_path, safe: Asciidoctor::SafeMode::SAFE)
       end
-      assert_match(/Failed to load AsciiDoc document/, exception.message)
+      # NOTE truffleruby cannot throw wrapped exception from rescue block of method
+      expected_message = RUBY_ENGINE == 'truffleruby' ? 'invalid byte sequence in UTF-8' : 'Failed to load AsciiDoc document'
+      assert_includes exception.message, expected_message
```

Comment:
Is this something we should be fixing in TruffleRuby? Is there an associated issue?

Reply:
I'm going to follow up by creating issues with test scenarios. Stay tuned.

Reply:
see https://github.com/oracle/truffleruby/issues/1542